framework/src/main/java/com/jfcbuilder/builders/IXYAnnotationBuilder.java

```diff
@@ -126,4 +43,21 @@
    *         called
    */
   XYAnnotation build() throws IllegalStateException;
+
+  /**
+   * Helper mutation method for replacing the configured x-axis date values with corresponding
+   * source array element index value.
+   * <p>
+   * Meant to be called internally by the framework itself, not by framework clients.
+   * <p>
+   * For facilitating the removal of visible time gaps on charts.
+   * <p>
+   * Different implementations may use different methods to map configured x value(s) to
+   * numeric indices, including search algorithms that may have higher order Big-O complexity. 
+   * 
+   * @param timeData             The array of source time values
+   * @param indexRangeStartIndex The start index of the series index range
+   * @param indexRangeEndIndex   The end index of the series index range
+   */
+  void mapXToTimeIndex(long[] timeData, int indexRangeStartIndex, int indexRangeEndIndex);
```

Comment:
I didn't want to put this in the interface but it was needed to invoke implementation-specific logic by the framework itself. I added some guidance on that in the documenting Javadoc.